### PR TITLE
Initial version of filebeat

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+|--------------------|-----------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `filebeat_config`  | `{ path: { logs: /var/log/filebeat, data: /var/lib/filebeat }}` | Since the filebeat settings file is a yaml file we write the whole object to the settings file. For more Information please see the [filebeat settings file documentation](https://www.elastic.co/guide/en/filebeat/current/filebeat-settings-file.html) |
+| `filebeat_modules` | `{}`                                                            | Dictionary of config files to write to modules.d/. More information below.                                                                                                                                                                               |
+
+### `filebeat_modules`
+Each entry in the `filebeat_modules` consits out of the following entries.
```
Comment (SlothOfAnarchy) [inline README.md]: *consists

File: README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| Variable | Default / Mandatory | Description                                                                              |
+|----------|---------------------|------------------------------------------------------------------------------------------|
+| `name`   | :heavy_check_mark:  | Name of the module config file. This value has to be the same as the name of the module. |
```
Comment (SlothOfAnarchy) [inline README.md]: `filebeat_modules` is a dictionary according to readme. What key shall be used?

Why not use the module name as key and its configuration as the corresponding value.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+## Requirements
+
+An apt based linux system
+
+## Role Variables
+
+
+| Variable           | Default / Mandatory                                             | Description                                                                                                                                                                                                                                              |
+|--------------------|-----------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `filebeat_config`  | `{ path: { logs: /var/log/filebeat, data: /var/lib/filebeat }}` | Since the filebeat settings file is a yaml file we write the whole object to the settings file. For more Information please see the [filebeat settings file documentation](https://www.elastic.co/guide/en/filebeat/current/filebeat-settings-file.html) |
```
Comment (SlothOfAnarchy) [inline README.md]: This is not the default that is actually used.

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+
+- name: Install apt-transport-https
+  become: yes
+  apt: 
+    name: "{{ item }}"
+  with_items:
+    - apt-transport-https
+
+- name: Add Elasticsearch repository key
```
Comment (SlothOfAnarchy) [inline tasks/main.yml:9]: why?

File: tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+
+- name: Install apt-transport-https
+  become: yes
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: still the default

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 4 above)
Comment (Scriptkiddi) [reply tasks/main.yml:9]: Because everything from the elk stack is signed with the elasticsearch key

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: tasks/main.yml (hunk 5 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: fixed

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy pls rereview

File: README.md (hunk 2 above)
Comment (SlothOfAnarchy) [reply README.md]: But why not use the dict for it? It worked in your other role.

Comment (Scriptkiddi) [reply README.md]: fixed

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy  pls rereview

File: README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+| `filebeat_config`  | `{ output.elasticsearch: { hosts: ["localhost:9200"] }, filebeat.config.modules: { path: ${path.config}/modules.d/*.yml , reload.enabled: false } }` | Since the filebeat settings file is a yaml file we write the whole object to the settings file. For more Information please see the [filebeat settings file documentation](https://www.elastic.co/guide/en/filebeat/current/filebeat-settings-file.html) |
+| `filebeat_modules` | `{}`                                                                                                                                                 | Dict of config files to write to modules.d/. More information below.                                                                                                                                                                                     |
+
+### `filebeat_modules`
+Each entry in the `filebeat_modules` consists out of the following entries.
+The key of each entry is used to determine the name of the config file and the content of the entry is writen to the config file
```
Comment (SlothOfAnarchy) [inline README.md]: *written
`.` to conclude the sentence
:P

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (Scriptkiddi) [reply README.md]: fixed

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy pls rereview

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**